src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs

```diff
@@ -358,45 +353,72 @@ public Result Execute(ExternalCommandData commandData, ref string message, Eleme
         }
     }
 
-    internal class Utils
+    /// <summary>
+    /// VersionSelectorData: A class to keep the selected version data to be stored in registry.
+    /// </summary>
+    internal class VersionSelectorData
     {
-        internal static void WriteToFile(string loadPath, string versionName)
-        {
-            var path = GetVersionSaveFileLocation(versionName);
-
-            if (File.Exists(path))
-            {
-                File.Delete(path);
-            }
-
-            using (var tw = new StreamWriter(path))
-            {
-                tw.WriteLine(loadPath);
-            }
+        const string RegKey64 = @"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\";
+        
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public VersionSelectorData()
+        { 
+            ThisDynamoVersion = Assembly.GetExecutingAssembly().GetName().Version;
         }
 
         /// <summary>
-        /// Return PreferenceSettings Default XML File Path if possible
+        /// Reads the version selection data for specific revit version from 
+        /// registry and returns a new VersionSelectorData.
         /// </summary>
-        internal static string GetVersionSaveFileLocation(string versionName)
+        /// <param name="revitVersion">Revit version</param>
+        /// <returns>VersionSelectorData</returns>
+        public static VersionSelectorData ReadFromRegistry(string revitVersion)
         {
-            try
-            {
-                string appDataFolder = System.Environment.GetFolderPath(
-                    System.Environment.SpecialFolder.ApplicationData);
+            var data = new VersionSelectorData() { RevitVersion = revitVersion };
+            var regKey = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry64);
+            regKey = regKey.OpenSubKey(RegKey64);
+            
+            var key = string.Format(@"Dynamo {0}.{1}", data.ThisDynamoVersion.Major, data.ThisDynamoVersion.Minor);
```

Comment:
Agree, there will be a different entry for every version of Dynamo installed, but these will be removed when Dynamo gets uninstalled.